index.js

```diff
@@ -285,17 +285,8 @@ Buffer.compare = function compare (a, b) {
   var x = a.length
   var y = b.length
 
-  var i = 0
-  var len = Math.min(x, y)
-  while (i < len) {
-    if (a[i] !== b[i]) break
-
-    ++i
-  }
-
-  if (i !== len) {
-    x = a[i]
-    y = b[i]
+  for (var i = 0, len = Math.min(x, y); i < len; ++i) {
+    if (a[i] !== b[i]) return a[i] > b[i] ? 1 : -1
```

Comment:
Also, I think that early returns were intentionally avoided in this function for perf reasons. Going to remove it. But your other changes are fine.